src/DataService/NotificationListener.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Lullabot\Mpx\DataService;
+
+use Lullabot\Mpx\Service\AccessManagement\ResolveAllUrls;
+use Lullabot\Mpx\Service\AccessManagement\ResolveDomain;
+use Lullabot\Mpx\Service\IdentityManagement\UserSession;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Listen to events from MPX.
+ *
+ * @see https://docs.theplatform.com/help/wsf-subscribing-to-change-notifications
+ *
+ * @codeCoverageIgnore Until we refactor this to allow loading all objects.
+ */
+class NotificationListener
+{
+    /**
+     * The resolver for MPX services.
+     *
+     * @var ResolveDomain
+     */
+    protected $resolveDomain;
+
+    /**
+     * The notification URL.
+     *
+     * @var string
+     */
+    protected $uri;
+
+    /**
+     * The user session to use for authenticated requests.
+     *
+     * @var UserSession
+     */
+    protected $session;
+
+    /**
+     * The name of the service to listen to notifications on, such as 'Media Data Service.'.
```

Comment:
Is the ending dot included in the name of the service? You've written 'Media Data Service.'

Reply:
No, it's not, good catch!